### PR TITLE
Add contribution guideline

### PR DESCRIPTION
**Pull request by elkoniu**
This is migration of old documentation including:
https://sssd.io/docs/developers/contribute.html
https://sssd.io/docs/newcomers/getting_started.html#github-workflow

### CODE REVIEW

**Comment by elkoniu** [comment]
I am not sure if bug report is in the scope of migration as it refers mostly "before GitHub era":
https://sssd.io/docs/users/reporting_bugs.html
If it should be migrated I think it should be part of contribution guideline and I can include it in this PR.
@pbrezina what do you think?

**Comment by elkoniu** [comment]
One thing I do not included in this PR is "Notes for Maintainers" section from:
https://sssd.io/docs/newcomers/getting_started.html#github-workflow
I am not sure if this should be part of contributor-facing documentation, please advice here.

File: src/contrib/contribution-policy.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Contribution policy
+===================
+
+There are many ways you can contribute to the System Security Services Daemon.
+This page should provide some basic pointers.
```
Comment (justin-stephenson) [inline src/contrib/contribution-policy.rst:5]: I think this line can be removed, s/This page should provide some basic pointers//

**Comment by justin-stephenson** [comment]
Besides the `Tasks for newcomers` section, the contribution-policy document does not actually contain any information about contributing. Maybe the github-workflow content should be merged into this doc.

**Comment by justin-stephenson** [comment]
Regarding the github-workflow content, I wonder if we really need to provide all of the github commands involved in the github fork & pull request workflow, IMO we could clean up the document if we only include the noteworthy SSSD specific steps/information, for example we would keep:

`Before submitting a patch, always make sure it doesn’t break SSSD tests and applies to the latest upstream master branch.`

But we don't really need `Setting up git` or `Downloading the code` sections, that information can be found in several other places with arguably better github fork + PR workflow information.

File: src/contrib/contribution-policy.rst (hunk 1 above)
Comment (elkoniu) [reply src/contrib/contribution-policy.rst:5]: Fixed

**Comment by elkoniu** [comment]
@justin-stephenson Small explanation from my side is needed I think.

Scope of this PR was to migrate current contribute documentation [1].
I found GitHub workflow document [2] which overlaps with [1] so I merged it in. At the end I split combined [1] and [2] content into few files to make it more clean and easier to update later.

Current state is that nothing is missing from old documentation - just the form is different.
Maybe my way of splitting this into multiple files was wrong - I do not know.
I agree with you that it needs to be refactored. For example we should assume that developer is familiar with git / GitHub. If not - the same way we should provide C programming tutorial etc.
I think documentation refining will be the next step after migration will be completed.

[1] https://sssd.io/docs/developers/contribute.html
[2] https://sssd.io/docs/newcomers/getting_started.html#github-workflow

**Comment by pbrezina** [comment]
> I agree with you that contribute guideline needs to be refactored. For example we should assume that developer is familiar with git / GitHub. If not - the same way we should provide C programming tutorial etc.

Don't assume, but provide link to existing github documentation instead of providing yet another one.

> I think documentation refining will be the next step after migration will be completed.

Yes. But if there's already something that you'd like to do, feel free to do it.

**Comment by pbrezina** [comment]
I talked to Pavel and he has no plan to further update this document. Ack.

We will revisit the document later when we finish migration of old  documents.

**Pull request merged**